src/libraries/System.Private.CoreLib/src/System/Text/Ascii.Equality.cs

```diff
@@ -540,24 +592,68 @@ public static Vector512<ushort> Load512(ref byte ptr)
             }
 
             [MethodImpl(MethodImplOptions.AggressiveInlining)]
-            [CompExactlyDependsOn(typeof(Avx))]
-            public static bool EqualAndAscii256(ref byte utf8, ref ushort utf16)
+            public static bool EqualAndAscii64(ref byte utf8, ref ushort utf16)
             {
+                Debug.Assert(Vector128.IsHardwareAccelerated);
```

Comment:
In the case that in L51 the Vec128.IsHardwareAccelarated = false this would blow-up here. There's really a test-hole?!

But when `ILoader.Use64Bit` (see above) take the HW into account it's safe.